nameko_vault/__init__.py

```diff
@@ -3,26 +3,37 @@
 
 
 class VaultProvider(DependencyProvider):
+    def __init__(self, mount_point=None):
+        self.mount_point = mount_point
+
     def setup(self):
         url = self.container.config.get("VAULT_URL", "")
         self.client = hvac.Client(url=url)
-        if self.client.is_sealed():
-            keys = self.container.config.get("VAULT_KEYS", "")
-            self.client.sys.submit_unseal_keys(
-                keys.split(",")
-            )
-
         if self.client.is_authenticated() is False:
             self.client.token = self.container.config.\
                 get("VAULT_TOKEN", "")
 
-    def get_kv_secret(self, mount_point="", path=""):
+    def get_kv_secret(self, path, mount_point=None):
+        mount_point = mount_point if mount_point else self.mount_point
         try:
             secret = self.client.secrets.kv.\
                 read_secret_version(mount_point=mount_point, path=path)
             return secret["data"]
         except Exception:
             return None
 
+    def get_kv_secrets_list(self, path, mount_point=None):
+        mount_point = mount_point if mount_point else self.mount_point
+        try:
+            secret = self.client.secrets.kv.\
+                list_secrets(path=path, mount_point=mount_point)
+            path = path if path.endswith("/") else path + "/"
+            return list(map(lambda key: path + key, secret["data"]["keys"]))
```

Comment:
Sempre que possível dar preferência para _list comprehension_ ao invés de `map` e/ou `filter` com `lambda`:
```suggestion
            return [path + key for key in secret["data"]["keys"]]
```

Geralmente executa mais rápido e é mais comum de se encontrar em código Python.

Aqui tem um artigo curto com algumas curiosidades sobre isso:
https://www.artima.com/weblogs/viewpost.jsp?thread=98196

Havia até uma intenção de remover essas funções do Python 3.